Compilation error sml.hpp

I tried to install and compile a program with SML and I get this error

**[]/include/boost/sml.hpp:166:8: error: previous definition of ‘struct boost::sml::v1_1_0::aux::function_traits<R (T::*)(TArgs ...)>’**

Have I done something wrong?

--- a/include/boost/sml.hpp
+++ b/include/boost/sml.hpp
@@ -170,7 +170,7 @@ template <class R, class T, class... TArgs>
 struct function_traits<R (T::*)(TArgs...) const> {
   using args = type_list<TArgs...>;
 };
-#if __cplusplus > 201402L
+#if __cplusplus > 201402L && __cpp_noexcept_function_type >= 201510
 template <class R, class... TArgs>
 struct function_traits<R (*)(TArgs...) noexcept> {
   using args = type_list<TArgs...>;

--- a/include/boost/sml/aux_/type_traits.hpp
+++ b/include/boost/sml/aux_/type_traits.hpp
@@ -139,7 +139,7 @@ template <class R, class T, class... TArgs>
 struct function_traits<R (T::*)(TArgs...) const> {
   using args = type_list<TArgs...>;
 };
-#if __cplusplus > 201402L  // __pph__
+#if __cplusplus > 201402L && __cpp_noexcept_function_type >= 201510  // __pph__
 template <class R, class... TArgs>
 struct function_traits<R (*)(TArgs...) noexcept> {
   using args = type_list<TArgs...>;

